partitions_test.go

```diff
@@ -189,3 +189,42 @@ func TestPartitionSpecToPath(t *testing.T) {
 	assert.Equal(t, "my%23str%25bucket=my%2Bstr/other+str%2Bbucket=%28+%29/my%21int%3Abucket=10",
 		spec.PartitionToPath(record, schema))
 }
+
+func TestGetPartitionFieldName(t *testing.T) {
+	schema := iceberg.NewSchema(0,
+		iceberg.NestedField{ID: 1, Name: "str", Type: iceberg.PrimitiveTypes.String},
+		iceberg.NestedField{ID: 2, Name: "other_str", Type: iceberg.PrimitiveTypes.String},
+		iceberg.NestedField{ID: 3, Name: "int", Type: iceberg.PrimitiveTypes.Int32, Required: true})
+
+	field := iceberg.PartitionField{
+		SourceID: 1, FieldID: 1000,
+		Transform: iceberg.TruncateTransform{Width: 19}, Name: "random_name",
+	}
+	name, err := iceberg.GeneratePartitionFieldName(schema, field)
+	assert.NoError(t, err)
+	assert.Equal(t, "str_trunc_19", name)
+
+	field = iceberg.PartitionField{
+		SourceID: 2, FieldID: 1001,
+		Transform: iceberg.BucketTransform{NumBuckets: 7}, Name: "another_random_name",
+	}
+	name, err = iceberg.GeneratePartitionFieldName(schema, field)
+	assert.NoError(t, err)
+	assert.Equal(t, "other_str_bucket_7", name)
+
+	field = iceberg.PartitionField{
+		SourceID: 2, FieldID: 1001,
+		Transform: iceberg.IdentityTransform{}, Name: "name",
+	}
+	name, err = iceberg.GeneratePartitionFieldName(schema, field)
+	assert.NoError(t, err)
+	assert.Equal(t, "other_str", name)
+
+	field = iceberg.PartitionField{
+		SourceID: 3, FieldID: 1001,
+		Transform: iceberg.VoidTransform{}, Name: "name",
+	}
+	name, err = iceberg.GeneratePartitionFieldName(schema, field)
+	assert.NoError(t, err)
+	assert.Equal(t, "int_null", name)
```

Comment:
Can we add tests for the rest of the transform types? Or at least one that hits the `default` condition?